Tools/autotest/common.py

```diff
@@ -1879,9 +1905,81 @@ def test_arm_feature(self):
                             self.set_rc(interlock_channel, 1000)
                             raise NotAchievedException("Motor interlock was changed while disarmed")
                 self.set_rc(interlock_channel, 1000)
+
+        self.start_test("Test all mode arming")
+        self.load_arming_test_mission()
+        try:
+            for mode in self.mav.mode_mapping():
+                self.start_test("Mode : %s" % mode)
+                if mode in self.NORMAL_ARMABLE_MODES_LIST:
+                    self.progress("Armable mode : %s" % mode)
+                    self.do_set_mode_via_command_long(mode)
+                    self.arm_vehicle()
+                    if not self.disarm_vehicle():
+                        self.progress("Failed to DISARM")
+                        raise NotAchievedException()
+                    self.progress("PASS arm mode : %s" % mode)
+                if mode in self.NOT_ARMABLE_MODES_LIST:
+                    if mode in self.NOT_SETTABLE_MODES_LIST:
+                        self.progress("Not settable mode : %s" % mode)
+                        try:
+                            self.do_set_mode_via_command_long(mode, timeout=2)
+                        except AutoTestTimeoutException:
+                            self.progress("PASS not arm mode : %s" % mode)
+                            pass
+                        except ValueError:
+                            self.progress("PASS not arm mode : %s" % mode)
+                            pass
+                    else:
+                        self.progress("Not armable mode : %s" % mode)
+                        self.do_set_mode_via_command_long(mode)
+                        self.run_cmd(mavutil.mavlink.MAV_CMD_COMPONENT_ARM_DISARM,
+                                     1,  # ARM
+                                     0,
+                                     0,
+                                     0,
+                                     0,
+                                     0,
+                                     0,
+                                     want_result=mavutil.mavlink.MAV_RESULT_FAILED
+                                     )
+                    self.mav.motors_disarmed_wait()
```

Comment:
If it can't be armed, why wait for disarm?

Reply:
surely a leftover, I will test without as it add a 1s delay to catch a heartbeat.